knowledge-graph/src/main/scala/io/renku/knowledgegraph/entities/model.scala

```diff
@@ -166,13 +167,31 @@ private object model {
         visibility:       projects.Visibility,
         date:             plans.DateCreated,
         keywords:         List[plans.Keyword],
-        maybeDescription: Option[plans.Description]
+        maybeDescription: Option[plans.Description],
+        workflowType:     Workflow.WorkflowType
     ) extends Entity {
       override type Name = plans.Name
       override type Date = plans.DateCreated
     }
 
     object Workflow {
+      sealed trait WorkflowType { self: Product =>
+        final def name: String = self.productPrefix.toLowerCase
+      }
+      object WorkflowType {
+        case object Composite extends WorkflowType
+        case object Basic     extends WorkflowType
+
+        val all: NonEmptyList[WorkflowType] =
+          NonEmptyList.of(Composite, Basic)
+
+        def fromName(str: String): Either[String, WorkflowType] =
+          all.find(_.name.equalsIgnoreCase(str)).toRight(s"Invalid workflowType name: $str")
+
+        implicit val encoder: Encoder[WorkflowType] =
+          Encoder.encodeString.contramap(_.name)
```

Comment:
Here I just choose a simple enum (string) for transporting the value to the consumer of the api. But open for other ways :)-

Reply:
Sure, I'd maybe name it `Step` instead of `Basic` for better consistency in naming.